### PR TITLE
Add Eclipse/J-Link debugging docs

### PR DESCRIPTION
**Pull request by hamishwillee**
@dinomani Can you review? There are some open questions below.

There is a rendered version of this here: https://hamishwillee.gitbooks.io/havdevguide/content/v/bf78793716f0044bc4ebb4deca75554ca31f0f60_6/en/debug/eclipse_jlink.html

### CODE REVIEW

File: en/debug/eclipse_jlink.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# MCU Eclipse/J-Link Debugging for PX4
+
+This topic explains how to setup and use [MCU Eclipse](https://gnu-mcu-eclipse.github.io/) with a *Segger Jlink adapter* to debug PX4 based-devices.
```
Comment (hamishwillee) [inline en/debug/eclipse_jlink.md]: this says to debug "PX4" based devices, but  does this mean Pixhawk devices, or specifically PX4 or NuttX?

File: en/debug/eclipse_jlink.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+## Required Hardware
+
+- [J-Link EDU Mini](https://www.segger.com/products/debug-probes/j-link/models/j-link-edu-mini/)
+- Pixhawk Debug Adapter Board
```
Comment (hamishwillee) [inline en/debug/eclipse_jlink.md]: Where do we get this. HOw is it attached?

File: en/debug/eclipse_jlink.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## First Use
+
+1. Run *Eclipse*.
+1. Add a source by choosing **File > Import > C/C++ > Existing Code as Makefile Project** and click **Next**.
+1. Point it to the Firmware folder and give it a name, then select *ARM Cross GCC* in the *Toolchain for Indexer Settings* and click **Finish**.
+   Import takes a while, wait for it to finish
+1. Set the MCU settings: right-click on the top-level project in the Project Explorer, select “Properties” then under MCU choose SEGGER J-Link Path. Set it like the screenshot below shows.
+   ![Eclipse: Segger J-Link Path](../../assets/debug/eclipse_segger_jlink_path.png)
+1. Update packs:
```
Comment (hamishwillee) [inline en/debug/eclipse_jlink.md:47]: Note all these three steps are associated with updating packs - is that correct?>

File: en/debug/eclipse_jlink.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+To install the *Segger Jlink* tools:
+1. Download and run the [J-Link Software and Documentation Pack](https://www.segger.com/downloads/jlink/#J-LinkSoftwareAndDocumentationPack) for your OS (Windows and Linux packages available).
+   - On Linux the tools are installed in **usr/bin**.
+
+For more information, see: https://gnu-mcu-eclipse.github.io/debug/jlink/install/
+
+## First Use
+
+1. Run *Eclipse*.
```
Comment (hamishwillee) [inline en/debug/eclipse_jlink.md:40]: Do we have to connect up the vehicle before this? Is there a particular port we need to use for the JLink connection?

File: en/debug/eclipse_jlink.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+     > **Note** Remember that you must build the target from the command line before starting a debug session.
+     
+   ![Eclipse: GDB Segger Debug config](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config.png)
+1. The *Debugger* and *Startup* tabs shouldn’t need any modifications (just verify your settings with the screenshots below)
+
+   ![Eclipse: GDB Segger Debug config: debugger tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_debugger_tab.png)
+   ![Eclipse: GDB Segger Debug config: startup tab](../../assets/debug/eclipse_settings_debug_config_gdb_segger_build_config_startup_tab.png)
+
+
+## Appendix
```
Comment (hamishwillee) [inline en/debug/eclipse_jlink.md]: When do we need to do this stuff in the appending? Ie in what of the step above does this become important? If it is always important, should it be embedded in those steps?

File: en/debug/eclipse_jlink.md (hunk 1 above)
Comment (dinomani) [reply en/debug/eclipse_jlink.md]: Actually ever device running px4 and having an exposed debug port. This enables you to debug px4 firmware and nuttx.

File: en/debug/eclipse_jlink.md (hunk 2 above)
Comment (dinomani) [reply en/debug/eclipse_jlink.md]: This debug adapter board is currently only available internally. So we may be just call it an adapter to connect segger jlink to debug port.

File: en/debug/eclipse_jlink.md (hunk 3 above)
Comment (dinomani) [reply en/debug/eclipse_jlink.md:47]: yes

File: en/debug/eclipse_jlink.md (hunk 4 above)
Comment (dinomani) [reply en/debug/eclipse_jlink.md:40]: Yes defentley, yo need to power your board and connect jlink to the host and the target(px4 running device)

File: en/debug/eclipse_jlink.md (hunk 5 above)
Comment (dinomani) [reply en/debug/eclipse_jlink.md]: the stuff in the appendix is the more manual way to get the register view working. You need this when you can not find the target cpu in the package manager. for me, this was the case for stm f7 controller. do you think it deserves its own chapter?

Comment (hamishwillee) [reply en/debug/eclipse_jlink.md]: Sort of. I added it to a troubleshooting section.

**Comment by hamishwillee** [comment]
Thanks @dinomani. Sorry for delay. Merging now.

**Pull request merged**